### ISSUE TITLE
Manipulating segmentation layer containing annotations

### ISSUE THREAD

**Issue by jsiddhant** [google/neuroglancer#206]
We have a need for manipulating structures (represented as a segmentation layer (annotation) link below). Somewhat similar to manipulating the transformation matrix (rotation, scaling + translation) under the source tab.

I was wondering if you had any suggestions on how we should go about implementing this and what sort of Interface method (i.e. key bindings/drag and drop) should be used.

Thanks,

Link to the viewer with our data: https://neuroglancer-demo.appspot.com/#!%7B%22dimensions%22:%7B%22x%22:%5B4.6000000000000004e-7%2C%22m%22%5D%2C%22y%22:%5B4.6000000000000004e-7%2C%22m%22%5D%2C%22z%22:%5B0.00002%2C%22m%22%5D%7D%2C%22position%22:%5B7874.16552734375%2C2444.890625%2C184.0110321044922%5D%2C%22crossSectionScale%22:9.018178748790614%2C%22projectionOrientation%22:%5B0.0019906109664589167%2C-0.0022145232651382685%2C0.007156445644795895%2C0.9999699592590332%5D%2C%22projectionScale%22:23605.130455376657%2C%22layers%22:%5B%7B%22type%22:%22image%22%2C%22source%22:%22precomputed://https://mousebrainatlas-datajoint-jp2k.s3.amazonaws.com/precomputed/MD589_fullres%22%2C%22name%22:%22image%22%7D%2C%7B%22type%22:%22segmentation%22%2C%22source%22:%22precomputed://https://test-bucket-sid.s3.amazonaws.com/final_precomputed_volumes/MD589%22%2C%22tab%22:%22source%22%2C%22name%22:%22annotation%22%7D%5D%2C%22selectedLayer%22:%7B%22layer%22:%22annotation%22%2C%22visible%22:true%2C%22size%22:599%7D%2C%22layout%22:%224panel%22%7D

**Comment by jbms** [comment]
Can you say a bit more about what mean as far as "manipulating structures?"  Do you mean drawing/moving contours?

**Comment by jsiddhant** [comment]
Yes sorry for the lack of detail. Each 'structure' is represented as an id in the segmentation layer (from 1-52). These 3D structures are aligned to the brain image layer outside neuroglancer and then the image and 3D structure volumes are converted to precomputed format to view.

**Comment by jbms** [comment]
So you want to interactively manipulate an affine transformation to try to align the data to the structure?  Is this a single affine transformation for all the structures, or a separate one for each structure?

**Comment by jsiddhant** [comment]
Yes exactly. We want to have it both at a global level and structure level.

**Comment by jbms** [comment]
I don't fully understand your application, but I suspect you may need to do a non-rigid alignment, in which case you will need some separate code to actually compute the aligned version, since Neuroglancer can only perform affine transformations.

In general I would suggest you prototype something using the Python integration, where you click on corresponding points in the two volumes, and as point correspondences are added the Python code recomputes an aligned volume (or if you are limited to affine transformations, the new affine transformation parameters).

**Comment by jsiddhant** [comment]
Is there a way to manipulate the affine transformation matrix (image below) of the layer (one under the source tab that allows you to rotate and translate volumes) from the python integration? I looked through the code but couldn't exactly find what to use.
![image](https://user-images.githubusercontent.com/23035136/77299824-4d93e680-6caa-11ea-915d-1238fb852b98.png)


This will allow us to initially manipulate the render which seems to be much faster as compared to calculating the full new transformed volume and then updating the layer with it.

**Comment by stephenrparsons** [comment]
@jsiddhant do you know if you figured out the above (manipulating this via Python?)
